integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/TaskFhirResourceProviderIntegrationTest.java

```diff
@@ -0,0 +1,382 @@
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		org.hl7.fhir.r4.model.OperationOutcome operationOutcome = readOperationOutcome(response);
+		
+		assertThat(operationOutcome, notNullValue());
+		assertThat(operationOutcome.hasIssue(), is(true));
+	}
+	
+	@Test
+	public void shouldThrow404ForNonExistingTaskAsXML() throws Exception {
+		MockHttpServletResponse response = get("/Task/" + WRONG_TASK_UUID).accept(FhirMediaTypes.XML).go();
+		
+		assertThat(response, isNotFound());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+	}
+	
+	@Test
+	public void shouldCreateNewTaskAsJson() throws Exception {
+		String jsonTask;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(JSON_CREATE_TASK_DOCUMENT)) {
+			jsonTask = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+		// create task
+		MockHttpServletResponse response = post("/Task").accept(FhirMediaTypes.JSON).jsonContent(jsonTask).go();
+		
+		// verify created correctly
+		assertThat(response, isCreated());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Task task = readResponse(response);
+		
+		assertThat(task, notNullValue());
+		assertThat(task.getIdElement().getIdPart(), notNullValue());
+		assertThat(task.getAuthoredOn(), equalTo("2016-10-31T08:25:05+10:00"));
+		assertThat(task.getStatus(), equalTo("ACCEPTED"));
+		assertThat(task.getIntent(), equalTo("order"));
+		assertThat(task, validResource());
+		
+		assertThat(task.getOwner(), equalTo("owner Requested Task"));
+		assertThat(task, validResource());
+		
+		// try to get new task
+		response = get("/Task/" + task.getIdElement().getIdPart()).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		
+		Task newTask = readResponse(response);
+		
+		assertThat(newTask.getId(), equalTo(task.getId()));
+		assertThat(newTask.getIdElement().getIdPart(), notNullValue());
+	}
+	
+	@Test
+	public void shouldCreateNewTaskAsXML() throws Exception {
+		String xmlTask;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(XML_CREATE_TASK_DOCUMENT)) {
+			Objects.requireNonNull(is);
+			xmlTask = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+		// create test
+		MockHttpServletResponse response = post("/Task").accept(FhirMediaTypes.XML).xmlContext(xmlTask).go();
+		
+		// verify created correctly
+		assertThat(response, isCreated());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Task task = readResponse(response);
+		
+		assertThat(task, notNullValue());
+		assertThat(task.getIdElement().getIdPart(), notNullValue());
+		assertThat(task.getAuthoredOn(), equalTo("2016-10-31T08:25:05+10:00"));
```

Comment:
So this field will always be the date and time the task was created, regardless of what's in the XML / JSON. I would remove the `authoredOn` field from the test data and remove this assertion.